src/main/java/org/jenkinsci/plugins/workflow/steps/durable_task/DurableTaskStep.java

```diff
@@ -353,17 +423,67 @@ private void check() {
             }
         }
 
+        // called remotely from HandlerImpl
+        @Override public void exited(int exitCode, byte[] output) throws Exception {
+            try {
+                getContext().get(TaskListener.class);
```

Comment:
Why are we checking for TaskListener in the context in order to identify this condition?

I'll take my answer in the form of a code comment, please.

Reply:
ack

Reply:
(done)